articles/architecture-scenarios/implementation/b2c/b2c-authorization.md

```diff
@@ -12,26 +12,84 @@ useCase:
 
-* **Authentication**: the process of determining if the user is who they say they are.
-* **Authorization**: the proceess of determining what the user is allowed to do in the system.
-* **Access Control**: the process of limiting a user to only the actions permitted, based on a combination of who the user is, what they are allowed to do in the system, and their consent.
+* **Authentication**: the process of determining if a user is who they say they are.
+* **Authorization**: the proceess of determining what is allowed, based on who a user is and the consent they provide.
+* **Access Control**: the process of limiting the specific set of actions, based on the combination of who the user is, what is allowed, and the set of contextually specific access criteria.
 
-For application level Authorization - typically referred to as Access Control - custom claims can be added to an OpenID Connect (OIDC) [ID Token](/tokens/id-token) via use of Auth0’s Rule [extensibility mechanism](#id-token-claims), and you will need to decide what that information might be in order for your application to make access control decisions.
+Access Control is essentially a super-set of the authorization process; authorization in it's fundemantal form can be thought of as a coarse-grained operation, whereas Access Control provides much more fine-grained control. Whatever your use case, there are a number of things you will want to consider when looking at functionality and workflow:
+
+* Are there scenarios where access to an application or API should be rejected?
+* Will I be providing APIs that can be accessed by third-party applications?
+* Will my APIs also be accessed by our own (first-party) applications?
+* Will my application be calling a third-party API?
+* Should my applications and/or APIs be enforcing role or permission based access control?
+ 
+Auth0 provides coarse-grained authorization by restricting access to certain applications or APIs based on certain conditions. In a coarse-grained authorization scenario, a [Rule](/rules) can be built that returns an `UnauthorizedError` when, for example, a user attempts access to an application or an API at an incorrect time (as described in this [example](/authorization/concepts/sample-use-cases-rules#allow-access-only-on-weekdays-for-a-specific-application)) - or if the user doesn’t have the right claim(s) contained in their [Metadata](/users/concepts/overview-user-metadata). For an _application_ using OpenID Connect ([OIDC](/protocols/oidc)), this would prevent the allocation of the [ID Token](/tokens/id-token) used to authorize access. Similarly, for an _API_, allocation of any OAuth2 [Access Tokens](/tokens/overview-access-tokens) (used when [calling the API](/api-auth/why-use-access-tokens-to-secure-apis)), could be prevented as described in this [example](/api-auth/restrict-access-api#example-deny-access-to-anyone-calling-the-api).
```

Comment:
We should point at app-metadata instead of user-metadata for a user not having the right claims(s), right?  Or are you saying if they haven't set a phone number yet, they can't access a particular app.

Reply:
My statement was referring to the fact that any metadata associated with a user could be used. We could qualify that this with something like: `app_metadata` for the stuff a user can't update and `user_metadata` for stuff they can?

Reply:
I've clarified using `app` metadata via best practice callout.